sxs/simulations/analyze.py

```diff
@@ -65,11 +68,9 @@ def compute_error_summary(wa, wb, t1, t2, modes=None, ASDs=None, total_masses=No
 
     wa_tilde = wa.fourier_transform()
     wb_tilde = wb.fourier_transform()
-    if ASDs is not None:
-        if total_masses is None:
-            raise ValueError("Need to specify total masses if ASDs are provided.")
-        for ASD in ASDs:
-            for total_mass in total_masses:
+    if ASDs_and_total_masses is not None:
+        for ASD, ASD_and_total_masses in ASDs_and_total_masses.items():
```

Comment:
Maybe change the names to be a little more descriptive, including unpacking:
```suggestion
        for ASD_name, (ASD_data, total_masses) in ASDs_and_total_masses.items():
```